cinema/views.py

```diff
@@ -118,21 +161,31 @@ class OrderPagination(PageNumberPagination):
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet
+):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication, )
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly, )
 
     def get_queryset(self):
         return Order.objects.filter(user=self.request.user)
 
     def get_serializer_class(self):
         if self.action == "list":
             return OrderListSerializer
-
         return OrderSerializer
 
     def perform_create(self, serializer):
         serializer.save(user=self.request.user)
+
+    def get_permissions(self):
+        if self.action in ("create", ):
+            return [IsAuthenticated()]
+        return super().get_permissions()
```

Comment:
Why do you need this code method?

Reply:
Because I should  give the ability for authenticated users to create order. 
And if i remove this method i get following error: 
Failure
Traceback (most recent call last):
  File "C:\Users\Expert\PycharmProjects\py-token-authentication\cinema\tests\test_order_api.py", line 65, in test_post_order
    self.assertNotEqual(response.status_code, status.HTTP_403_FORBIDDEN)
AssertionError: 403 == 403